pkg/watcher/rated.go

```diff
@@ -93,15 +105,56 @@ func getValidationStatistics(cfg *core.Config, key string) (*getValidatorEffecti
 			"status-code":    res.StatusCode,
 			"validation-key": key,
 			"nb-entries":     len(response.Data),
-		}).Warn("expected 1 entry of statistics for the validation key")
+		}).Warn("Expected 1 entry of statistics for the validation key")
 
-		return nil, fmt.Errorf("unexpected data response from rated network")
+		return nil, backoff.Permanent(fmt.Errorf("unexpected data response from rated network"))
 	}
 
 	log.WithFields(log.Fields{
 		"url":            url,
 		"validation-key": key,
-	}).Info("fetched validator statistics about key from rated network")
+	}).Info(fmt.Sprintf("Successfully fetched statistics for validator %s from rated network", key))
 
 	return &response.Data[0], nil
+
+}
+
+// getValidationStatistics fetches data about a validator on rated network.
+func getValidationStatistics(cfg *core.Config, key string) (*getValidatorEffectivenessData, error) {
+	// Key here is already prefixed with '0x', this is required for rated
+	// network API (which also supports indexes).
+	url := fmt.Sprintf("%s/v0/eth/validators/%s/effectiveness?size=1", cfg.ApiEndpoint, key)
+
+	log.WithFields(log.Fields{
+		"url":            url,
+		"validation-key": key,
+	}).Info("Fetching rated data for validation key")
+
+	var result *getValidatorEffectivenessData
+	operation := func() error {
```

Comment:
I totally appreciate how weird this is. In the Scala world I would get fired for this. Oddly, this is how it is supposed to be used.

Reply:
Yup, this library looks overly complicated, I'd probably have done it using a pattern like this (pseudo-code):

```
backoff := {1, 3, 7, ...}
for _, backoff := range backoff {
      resp, err := get(url)
      if err != nil {
           log...
           return err
      }

      if resp.statuscode == 200 {
         // do stuff
         return nil
      }

      log...
      time.Sleep(backoff)
}

return fmt.Errorf("giving up after a few retries...")
```

This makes it a bit more flexible and easier to maintain, I don't think go code should be 'optimized' for lines, it's verbose by nature but clear,  everything is in this place with all the context. I'd favor having 3/4 implems using such a pattern before considering abstracting or refactoring it.

Reply:
Hmm, I see what you mean. I would have to implement a jitter on top to avoid synchronising the retries. 
I will think about it, this is a bit more involved.